### PR TITLE
Fix crash in dashboard

### PR DESCRIPTION
**Pull request by miasb**

### CODE REVIEW

File: src/input_cp2k_mp2.F [hunk 1]
```diff
@@ -1042,6 +1043,59 @@ SUBROUTINE create_print_section(section)
       CALL section_add_subsection(section, print_key)
       CALL section_release(print_key)
 
+      CALL section_create(gw_dos_section, __LOCATION__, name="GW_DOS", &
+                          description="Section for printing the spectral funtion.", &
```
Comment (mkrack) [inline src/input_cp2k_mp2.F]: "function"

File: src/rpa_gw.F [hunk 2]
```diff
@@ -1389,6 +1391,23 @@ SUBROUTINE compute_QP_energies(vec_Sigma_c_gw, count_ev_sc_GW, gw_corr_lev_occ,
...
          E_CBM_SCF_beta = 1.0E3
 
+         ndos = 0
+         dos_precision = mp2_env%ri_g0w0%dos_prec
+         dos_upper_bound = mp2_env%ri_g0w0%dos_upper
+         dos_lower_bound = mp2_env%ri_g0w0%dos_lower
+
+         IF (dos_lower_bound >= dos_upper_bound) THEN
+            CALL cp_abort(__LOCATION__, "Invalid settings for GW_DOS calculation!")
+            dos_precision = 0_dp
```
Comment (mkrack) [inline src/rpa_gw.F]: What is the effect of these follow-up variable initialisations after a cp_abort?

File: src/rpa_gw.F [hunk 3]
```diff
@@ -1534,6 +1561,22 @@ SUBROUTINE compute_QP_energies(vec_Sigma_c_gw, count_ev_sc_GW, gw_corr_lev_occ,
...
+         IF (dos_precision /= 0.0_dp) THEN
+            IF (iunit > 0) THEN
+               CALL open_file('spectral.dat', unit_number=iunit, file_status="UNKNOWN", file_action="WRITE")
+               DO idos = 1, ndos
+                  ! 1/pi
+                  ! [1/Hartree] -> [1/evolt]
+                  WRITE (iunit, '(E17.10, E17.10)') (dos_lower_bound + REAL(idos - 1, KIND=dp)*dos_precision)*evolt, &
+                     vec_gw_dos(idos)/evolt*0.3183098861837907
+               END DO
+               CLOSE (iunit)
```
Comment (mkrack) [inline src/rpa_gw.F]: What's about using the matching close_file subroutine instead of CLOSE.

File: src/rpa_gw.F [hunk 4]
```diff
@@ -4332,6 +4393,78 @@ SUBROUTINE continuation_pade(vec_gw_energ, vec_omega_fit_gw, &
...
+          n_level_gw_ref <= max_level_self_energy .AND. iunit > 0) THEN
+
+         CALL open_file('self_energy_re_'//TRIM(string_level)//'.dat', unit_number=iunit, &
+                        file_status="UNKNOWN", file_action="WRITE")
+         DO idos = 1, ndos
+            omega_dos = dos_lower_bound + REAL(idos - 1, KIND=dp)*dos_precision
+            WRITE (iunit, '(F17.10, F17.10)') omega_dos*evolt, vec_sigma_real(idos)
+         END DO
+
+         CLOSE (iunit)
```
Comment (mkrack) [inline src/rpa_gw.F]: CALL close_file()

File: src/rpa_gw.F [hunk 5]
```diff
@@ -4332,6 +4393,78 @@ SUBROUTINE continuation_pade(vec_gw_energ, vec_omega_fit_gw, &
...
+
+         CALL open_file('self_energy_im_'//TRIM(string_level)//'.dat', unit_number=iunit, &
+                        file_status="UNKNOWN", file_action="WRITE")
+         DO idos = 1, ndos
+            omega_dos = dos_lower_bound + REAL(idos - 1, KIND=dp)*dos_precision
+            WRITE (iunit, '(F17.10, F17.10)') omega_dos*evolt, vec_sigma_imag(idos)
+         END DO
+
+         CLOSE (iunit)
```
Comment (mkrack) [inline src/rpa_gw.F]: See above

File: src/rpa_gw.F [hunk 6]
```diff
@@ -1534,6 +1561,22 @@ SUBROUTINE compute_QP_energies(vec_Sigma_c_gw, count_ev_sc_GW, gw_corr_lev_occ,
...
+         iunit = cp_logger_get_default_unit_nr()
+
+         IF (dos_precision /= 0.0_dp) THEN
+            IF (iunit > 0) THEN
+               CALL open_file('spectral.dat', unit_number=iunit, file_status="UNKNOWN", file_action="WRITE")
+               DO idos = 1, ndos
+                  ! 1/pi
+                  ! [1/Hartree] -> [1/evolt]
+                  WRITE (iunit, '(E17.10, E17.10)') (dos_lower_bound + REAL(idos - 1, KIND=dp)*dos_precision)*evolt, &
+                     vec_gw_dos(idos)/evolt*0.3183098861837907
```
Comment (JWilhelm) [inline src/rpa_gw.F]: You could also think of using the "pi" variable from the physcon file.

File: src/rpa_gw.F [hunk 7]
```diff
@@ -4332,6 +4393,78 @@ SUBROUTINE continuation_pade(vec_gw_energ, vec_omega_fit_gw, &
...
+            WRITE (iunit, '(F17.10, F17.10)') omega_dos*evolt, vec_sigma_real(idos)
+         END DO
+
+         CLOSE (iunit)
+
+         CALL open_file('self_energy_im_'//TRIM(string_level)//'.dat', unit_number=iunit, &
+                        file_status="UNKNOWN", file_action="WRITE")
+         DO idos = 1, ndos
+            omega_dos = dos_lower_bound + REAL(idos - 1, KIND=dp)*dos_precision
+            WRITE (iunit, '(F17.10, F17.10)') omega_dos*evolt, vec_sigma_imag(idos)
```
Comment (JWilhelm) [inline src/rpa_gw.F]: One could also think of printing vec_sigma_imag(idos) in the evolt unit.

**Comment by miasb** [comment]
Thanks for your suggestions. I incorporated them in my latest commit.

**Comment by miasb** [comment]
Resolved merge conflicts

**Pull request merged**